student-store-ui/src/components/CheckoutForm/CheckoutForm.jsx

```diff
@@ -0,0 +1,164 @@
+import * as React from "react"
+import "./CheckoutForm.css"
+import { useState, useEffect } from "react"
+import axios from "axios"
+
+export default function CheckoutForm({ shoppingCart, setShoppingCart }) {
+
+    const [checkoutActive, setCheckoutActive] = useState(false)
+    const [emailInput, setEmailInput] = useState('')
+    const [nameInput, setNameInput] = useState('')
+    const [namesForDisplay, setNameForDisplay] = useState('')
+    const [emailForDisplay, setEmailForDisplay] = useState('')
+    const [danger, setDanger] = useState(false)
+    const [missingInput, setMissingInput] = useState(false)
+    const [total, setTotal] = useState(0)
+    const [taxesFees, setTaxesFees] = useState(0)
+    const [subtotalDisplay, setSubtotalDisplay] = useState(0)
+
+    function handleCheckout() {
+        if (shoppingCart.length === 0) {
+          setDanger(true);
+          setMissingInput(false);
+        } else if (emailInput === '' || nameInput === '') {
+          setDanger(false);
+          setMissingInput(true);
+        } else {
+          setCheckoutActive(true);
+          setNameForDisplay(nameInput);
+          setEmailForDisplay(emailInput);
+          setEmailInput('');
+          setNameInput('');
+          setDanger(false);
+          setMissingInput(false);
+          
+        }
+        informationToBackend();
+      }
+
+    let errorItemQuantity =
+        <>
+            <p class="is-danger" >No cart or items in cart found to checkout.</p>
+        </>
+    let noError =
+
+        <>
+            <p class="is-danger" ></p>
+        </>
+
+    let errorMissingInfo =
+    <>
+    <p class="is-danger" >User info must include an email and name.</p>
+    </>
+
+    function handleExit() {
+        setShoppingCart([]) // finish shopping and emptying the cart to new shop 
+        setCheckoutActive(false)
+    }
+
+    function subtotal(shoppingCart) {
+        return shoppingCart.reduce((sum, item) => {
+            return (sum + (item.price * item.quantity))
+        }, 0);
+    }
+
+    const handleNameInput = (e) => {
+        setNameInput(nameInput => nameInput = e.target.value)
+    }
+
+    const handleEmailInput = (e) => {
+        setEmailInput(emailInput => emailInput = e.target.value)
+    }
+
```

Comment:
use one piece of form state (object with multiple keys) to reduce having multiple handleChange functions